src/commonMain/kotlin/org/mongodb/kbson/BsonObjectId.kt

```diff
@@ -145,7 +146,16 @@ public class BsonObjectId(
 
         /** Create a new BsonObjectId */
         public operator fun invoke(): BsonObjectId {
-            return BsonObjectId(getCurrentTimeInSeconds(), RANDOM_VALUE1, RANDOM_VALUE2, nextCounter())
+            return invoke(getCurrentTimeInSeconds())
+        }
+
+        /**
+         * Create a new BsonObjectId
+         *
+         * @param timestamp the timestamp in millis
+         */
+        public operator fun invoke(timestamp: Long): BsonObjectId {
+            return invoke((timestamp / MILLIS_IN_SECOND).toInt())
```

Comment:
Dividing by `MILLIS_IN_SECONDS` doesn't look right if the timestamp is already supposed to be in millis?

Reply:
That seems right, but line 149 does not, since it's invoking this function with a value that's already been converted to seconds.

Reply:
ObjectId uses the time in seconds but most time classes use millis since epoch - so this converts that to seconds since epoch.

The original method already has the time in seconds so doesn't need to convert.  Both call the new private helper on line 187.

Reply:
Ah yeah, I was reading it wrong 👍